tests/integration/Core/Framework/DataAbstractionLayer/Event/EntityLoadedEventFactoryTest.php

```diff
@@ -65,9 +67,11 @@ public function testCreate(): void
             (new LanguageEntity())->assign(['id' => $this->ids->create('l1'), '_entityName' => 'language']),
         ]));
 
-        $events = $this->entityLoadedEventFactory->create([$product], Context::createDefaultContext());
-        static::assertNotNull($events->getEvents());
-        $createdEvents = $events->getEvents()->map(fn (EntityLoadedEvent $event): string => $event->getName());
+        $events = $this->entityLoadedEventFactory->create([$product], Context::createDefaultContext())->getEvents();
+        static::assertNotNull($events);
+        static::assertContainsOnlyInstancesOf(EntityLoadedEvent::class, $events);
+        /** @var NestedEventCollection<EntityLoadedEvent<Entity>> $events */
```

Comment:
I also do not like, that we have a var annotation in tests :thinking:

Reply:
Its now needed as the closure given for `->map` expects a `NestedEvent`, the default type for `NestedEventCollection`. Only we know, that the `NestedEventCollection` is only full of `EntityLoadedEvent`.
It should be fine to add that var type, as its secured by `assertContainsOnlyInstancesOf`